/*
Copyright (C) 2010 Srivats P.

This file is part of "Ostinato"

This is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>
*/

import "protocol.proto";

package OstProto;
// Tcp
message Tcp {

    optional bool is_override_hdrlen = 1;    
    optional bool is_override_cksum = 2;    

    optional uint32        src_port = 3 [default = 8902];
    optional uint32        dst_port = 4 [default = 80];

    optional uint32        seq_num = 5 [default = 129018];
    optional uint32        ack_num = 6;

    optional uint32        hdrlen_rsvd = 7 [default = 0x50];
    optional uint32        flags = 8;

    optional uint32        window = 9 [default = 1024];
    optional uint32        cksum = 10;
    optional uint32        urg_ptr = 11;
}

extend Protocol {
    optional Tcp tcp = 140;
}

